runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractContextStreamView.java

```diff
@@ -307,7 +307,13 @@ private void updatePointer(final ILogData data) {
      */
     protected boolean processEntryForContext(final ILogData data) {
         if (data != null) {
-            final Object payload = data.getPayload(runtime);
+            try {
+                final Object payload = data.getPayload(runtime);
```

Comment:
Why don't you push this down to `LogData::getPayload` ? (since that's when deserialize is actually invoked)

Reply:
done.